tools/rna_tools/targetfinder/TargetFinder.xml

```diff
@@ -1,73 +1,236 @@
-    </requirements>
+<tool id='targetfinder' name='TargetFinder' version='@TOOL_VERSION@' profile='20.01'>
+    <description>plant small RNA target prediction tool</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro='edam' />
+    <expand macro='requirements' />
     <stdio>
-        <exit_code range="1:" />
+        <exit_code range='1:' />
     </stdio>
     <command><![CDATA[
         targetfinder_threads.pl
             -f $f
             -d $d
             -c $c
+            -p $output_format
             -t "\${GALAXY_SLOTS:-12}"
             $r
-            -o $output1
+            #if $output_format == 'classic'
+                -o $output_text
+            #else if $output_format == 'table'
+                -o $output_tabular
+            #else if $output_format == 'json'
+                -o $output_json
+            #else if $output_format == 'gff'
+                -o $output_gff
+            #end if
     ]]></command>
     <inputs>
-        <param label="Input small RNA sequences file" help="(-f) FASTA-format" name="f" type="data" format="fasta" />
-        <param label="Target sequence database file" help="(-d) FASTA-format" name="d" type="data" format="fasta" />
-        <param label="Prediction score cutoff value" help="(-c) DEFAULT = 4" name="c" type="float" value="4.0" />
-        <param label="Search reverse strand for targets?" help="(-r) Use this option if the database is genomic DNA." name="r" type="boolean" falsevalue="" truevalue="-r" checked="false" />
+        <param argument='-f' type='data' format='fasta' label='Input small RNA sequences file' help='FASTA-format' />
+        <param argument='-d' type='data' format='fasta' label='Target sequence database file' help='FASTA-format' />
+        <param argument='-c' type='float' value='4.0' label='Prediction score cutoff value' help='Predicted targets are printed out if they are equal to or lower than the cutoff score specified. DEFAULT = 4' />
+        <param argument='-c' type='float' value='4.0' label='Prediction score cutoff value' help='Predicted targets are printed out if they are equal to or lower than the cutoff score specified. DEFAULT = 4' />
+        <param argument='-r' type='boolean' falsevalue='' truevalue='-r' checked='false' label='Search reverse strand for targets?' help='Use this option if the database is genomic DNA.'  />
+        <param name='output_format' argument='-p' type='select' label='Output format' help='Output format for small RNA-target pairs (Default = classic).' >
+                <option value='classic'>Original TargetFinder base-pair format</option>
+                <option value='gff'>GFF</option>
+                <option value='json'>JavaScript Object Notation (JSON)</option>
+                <option value='table'>Tab-delimited format</option>
+        </param>
     </inputs>
     <outputs>
-        <data name="output1" format="data"/>
+        <data name='output_tabular' format='tabular'>
```

Comment:
You can also do it this way: https://github.com/bgruening/galaxytools/blob/dee6a6f4aa8cbee9f81692fc6f3a2e4ad1f3abf6/tools/rest_tool/rest_tool.xml#L192

Reply:
Great, nice.